content/en/blog/_posts/2022-01-19-Securing-Admission-Controllers.md

```diff
@@ -0,0 +1,42 @@
+
+From the threat model, we developed a set of security best practices which should be adopted to ensure that cluster operators can get the security benefits of admission controllers whilst avoiding any risks from using them.
+
+## Admission controllers and good practices for security
+
+From the threat model there were a couple of themes which emerged around how to ensure the security of admission controllers.
+
+### Secure webhook configuration
+
+It’s important to ensure that any security component in a cluster is well configured and admission controllers are no different here. There are a couple of security best practices to consider when using admission controllers
+
+* **Correctly configured TLS for all webhook traffic**. Communications between the API server and the admission controller webhook should be authenticated and encrypted to ensure that attackers who may be in a network position to view or modify this traffic cannot do so. To achieve this access the API server and webhook must be using certificates from a trusted certificate authority so that they can validate their mutual identities
+* **Only authenticated access allowed**. If an attacker can send an admission controller large numbers of requests, they may be able to overwhelm the service causing it to fail. Ensuring all access requires strong authentication should mitigate that risk.
+* **Admission controller fails closed**. This is a security practice that has a tradeoff, so whether a cluster operator wants to configure it will depend on the cluster’s threat model. If an admission controller fails closed, when the API server can’t get a response from it, all deployments will fail. This stops attackers bypassing the admission controller by disabling it, but obviously can disrupt the cluster’s operation. As clusters can have multiple webhooks, one approach to hit a middle ground might be to have critical controls on a fail closed setups and less critical controls allowed to fail open.
+* **Regular reviews of webhook configuration**. Configuration mistakes can lead to security issues, so it’s important that the admission controller webhook configuration is checked to make sure the settings are correct. This kind of review could be done automatically by an Infrastructure As Code scanner or manually by an administrator.
+
+
+### Secure cluster configuration for admission control
+
+In most cases the admission controller webhook used by a cluster will be installed as a workload in the cluster. As a result, it’s important to ensure that Kubernetes security features which could impact its operation are well configured.
+
+* **Restrict RBAC rights**. Any user who has rights which would allow them to modify the configuration of the webhook objects or the workload that the admission controller uses, could disrupt its operation. So it’s important to make sure that only cluster administrators have those rights.
```

Comment:
```suggestion
* **Restrict RBAC rights**. Any user who has rights which would allow them to modify the configuration of the webhook objects or the workload that the admission controller uses could disrupt its operation. So it’s important to make sure that only cluster administrators have those rights.
```

Reply:
Manually added this one